### PR TITLE
Fix UFRM nodata handling

### PR DESCRIPTION
**Pull request by dcdenu4**
This PR fixes a bug where input rasters with nodata of 0 were not properly being masked and causing downstream side-effects. 

Closes issue #185

### CODE REVIEW

File: src/natcap/invest/urban_flood_risk_mitigation.py [hunk 1]
```diff
@@ -677,7 +677,7 @@ def _runoff_retention_op(q_pi_array, p_value, q_pi_nodata, result_nodata):
     result = numpy.empty_like(q_pi_array)
     result[:] = result_nodata
     valid_mask = numpy.ones(q_pi_array.shape, dtype=numpy.bool)
-    if q_pi_nodata:
+    if isinstance(q_pi_nodata, (int, float)):
```
Comment (richpsharp) [inline src/natcap/invest/urban_flood_risk_mitigation.py:680]: Ahhh i see, I feel like this is more obvious what's happening here if you could change this for a test for `None`like this:

```
if q_pi_nodata is not None:
```

This way it shows that `q_pi_nodata` is allowed to be None. Testing if it's a numeric seems like a circular way to do that and isn't obvious why it wouldn't be numeric.

File: src/natcap/invest/urban_flood_risk_mitigation.py [hunk 2]
```diff
@@ -702,7 +702,7 @@ def _q_pi_op(p_value, s_max_array, s_max_nodata, result_nodata):
 
     zero_mask = (p_value <= lam * s_max_array)
     non_nodata_mask = numpy.ones(s_max_array.shape, dtype=numpy.bool)
-    if s_max_nodata:
+    if isinstance(s_max_nodata, (int, float)):
```
Comment (richpsharp) [inline src/natcap/invest/urban_flood_risk_mitigation.py:705]: Same here, can you `if s_max_nodata is not None:`?

File: src/natcap/invest/urban_flood_risk_mitigation.py [hunk 3]
```diff
@@ -731,7 +731,7 @@ def _s_max_op(cn_array, cn_nodata, result_nodata):
     result[:] = result_nodata
     zero_mask = cn_array == 0
     valid_mask = ~zero_mask
-    if cn_nodata:
+    if isinstance(cn_nodata, (int, float)):
```
Comment (richpsharp) [inline src/natcap/invest/urban_flood_risk_mitigation.py:734]: `if cn_nodata is not None:`

File: src/natcap/invest/urban_flood_risk_mitigation.py [hunk 4]
```diff
@@ -759,9 +759,9 @@ def _lu_to_cn_op(
     result = numpy.empty_like(lucode_array, dtype=numpy.float32)
     result[:] = cn_nodata
     valid_mask = numpy.ones(lucode_array.shape, dtype=numpy.bool)
-    if lucode_nodata:
```
Comment (richpsharp) [inline src/natcap/invest/urban_flood_risk_mitigation.py:762]: `if lucode_nodata is not None:`

File: src/natcap/invest/urban_flood_risk_mitigation.py [hunk 5]
```diff
@@ -759,9 +759,9 @@ def _lu_to_cn_op(
     result = numpy.empty_like(lucode_array, dtype=numpy.float32)
     result[:] = cn_nodata
     valid_mask = numpy.ones(lucode_array.shape, dtype=numpy.bool)
-    if lucode_nodata:
+    if isinstance(lucode_nodata, (int, float)):
         valid_mask[:] &= ~numpy.isclose(lucode_array, lucode_nodata)
-    if soil_type_nodata:
+    if isinstance(soil_type_nodata, (int, float)):
```
Comment (richpsharp) [inline src/natcap/invest/urban_flood_risk_mitigation.py:764]: `if soil_type_nodata is not None:`?

**Comment by dcdenu4** [comment]
@richpsharp that's a good point and way less confusing. Fixed in https://github.com/natcap/invest/pull/231/commits/6549d3ed214e6f840d243bb1a3c68c590e563a59

**Pull request merged**